DisguiseUnityRenderStream/Runtime/DisguiseRenderStreamWithClustercs.cs

```diff
@@ -150,81 +156,180 @@ public static ClusterParams ProcessClusterParams(ClusterParams clusterParams)
-            else if (maybeRepeaterCount.HasValue)
+            catch (Exception e)
             {
-#endif
-                clusterParams.RepeaterCount = maybeRepeaterCount.Value;
+                Debug.LogException(e);
+                clusterParams.ClusterLogicSpecified = false;
+            }
+
+            return clusterParams;
+        }
 
-                // If we're running several nodes on the same machine, use a single-access file to keep track
-                // of node ids in use.
-                var retries = 0;
-                var nodeIdFilePath = Path.Combine(Path.GetTempPath(), Application.productName + ".node");
-                while (!detectedNodeId && retries < 10)
+        /// <summary>
+        /// Negotiate a distinct ID amongst other running render nodes. Uses the same multi-cast end point
+        /// as the core Cluster Display logic.
+        /// </summary>
+        /// <param name="clusterParams">Parameters containing the networking information.</param>
+        /// <returns></returns>
+        /// <exception cref="TimeoutException">The operation timed out.</exception>
+        /// <exception cref="OperationCanceledException">Multiple errors were encountered. See the log for details.</exception>
+        static byte NegotiateNodeID(ClusterParams clusterParams)
+        {
+            var address = IPAddress.Parse(clusterParams.MulticastAddress);
+            var sendEndPoint = new IPEndPoint(address, clusterParams.Port);
+            var timeoutMilliseconds = (int)clusterParams.HandshakeTimeout.TotalMilliseconds;
+            
+            using var udpClient = new UdpClient();
+            udpClient.EnableMulticast(address,
+                clusterParams.Port,
+                clusterParams.AdapterName,
+                timeoutMilliseconds);
+
+            int nodeIdResult;
+            var expectedNodeCount = clusterParams.RepeaterCount + 1;
+            
+            // Message that indicates that we're available to join a group.
+            // Each node must have a unique announcement message.
+            var announceMessage = $"announce:{Environment.MachineName}{Process.GetCurrentProcess().Id}";
+            
+            // Message that indicates that we're discovered a valid group.
+            var readyMessage = $"ready:{Environment.MachineName}{Process.GetCurrentProcess().Id}";
+            var foundGroup = false;
+            
+            // Set up our listening and announcing tasks.
+            var cancellation = new CancellationTokenSource();
+            var token = cancellation.Token;
+            
+            var listen = Task.Run(async () =>
+            {
+                // Listen for messages, including from self
+                SortedSet<string> announcements = new();
+                HashSet<string> readyReports = new();
+                while (!token.IsCancellationRequested)
                 {
-                    try
+                    var result = await udpClient.ReceiveAsync();
```

Comment:
Minor: Should we use the version with the cancellation token?

Reply:
Are you referring to the overload that accepts a token? Sadly, this version of .NET doesn't have it 😞

Reply:
Ho yes, I forgot, then too bad, keep it the way it is.